packages/sewing-kit-koa/src/assets.ts

```diff
@@ -167,18 +167,7 @@ function getAssetsFromManifest(
     return entrypointAssets;
   }
 
-  const bundleTester = new RegExp(`\\b${name}[^\\.]*\\.${kind}`);
```

Comment:
What's the context around this removal? @BPScott to allow any resource?

Reply:
I'm not really sure what this code was supposed to do (@lemonmade may have better insight)

It looks like it tries to say "inject the async assets before the first non vendor entrypoint" - so it goes VendorEntrypointAssets, AsnycAssets,  OtherEntryPointAssets" but there's a bug in it because findIndex returns -1 when not found so I think actually happens is

- When no vendor items found (findIndex returns -1) it injects the async assets at the last-but-one position in the EntryPoint Array (so SomeOtherEntrypointAssets, AsyncAssets, TheFinalEntrypointAsset) 
- When a vendor item is found as the first entrypointAsset (findIndex returns 0) it returns the async assets and then the entrypoint assets (so AsyncAssets, VendorEntryPointAssets, OtherEntryPointAssets)
- When greater than one vendor item is found it injects the Asnync assets before the first non vendor entrypoint asset (so VendorEntrypointAssets AsyncAssets, OtherEntrypointAssets).

With this PR we always put the async assets after the entrypoint ones (regardless of if those are vendor entrypoints or not) rather than trying to inject them in the middle so there's no need to do any of that slicing or dicing.

- Add the

Reply:
My assumption is that this would not work for JS, but I unfortunately don't have time to verify. My understanding was that webpack needed all of the async scripts to be loaded before the entrypoint that consumes them in order for those async assets to be resolvable synchronously from the entrypoint. The ability to resolve them synchronously is key since that is typically necessary to render the same screen as what was rendered during SSR.

Reply:
(Agreed that there is a bug here, though probably one that is effectively never hit given that vendor bundles are created fairly consistently; also agreed that this logic is correct for styles)